### PR TITLE
[fix](planner) fix pr "using crchash replace murmurhash in the runtime filter"

### PR DESCRIPTION
**Pull request by Mryange**
# Proposed changes
The past mistakes have beenRevert. [(https://github.com/apache/doris/pull/18730)]

## Problem summary

In the past code, there is a missing "else".
mNXXtmQio9
This causes tpcds_sf1_p1_q85 failed

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/exprs/bloom_filter_func.h [hunk 1]
```diff
@@ -463,6 +513,15 @@ class BloomFilterFunc final : public BloomFilterFuncBase {
         return dummy.find(*_bloom_filter, data);
     }
 
+    // This function is only to be used if the be_exec_version may be less than 2. If updated, please delete it.
+    bool find_crc32_hash(const void* data) const override {
+        if constexpr (std::is_same_v<typename BloomFilterTypeTraits<type>::FindOp, StringFindOp>) {
```
Comment (BiteTheDDDDt) [inline be/src/exprs/bloom_filter_func.h:518]: maybe `FixedStringFindOp` can use new hash also.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34 seconds
 stream load tsv:          427 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230419104045_clickbench_pr_131026.html

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p1

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/exprs/vbloom_predicate.cpp [hunk 2]
```diff
@@ -92,10 +92,20 @@ Status VBloomPredicate::execute(VExprContext* context, Block* block, int* result
...
     if (type.is_string_or_fixed_string()) {
-        for (size_t i = 0; i < sz; i++) {
-            auto ele = argument_column->get_data_at(i);
-            const StringRef v(ele.data, ele.size);
-            ptr[i] = _filter->find(reinterpret_cast<const void*>(&v));
+        // When _be_exec_version is equal to or greater than 2, we use the new hash method.
+        // This is only to be used if the be_exec_version may be less than 2. If updated, please delete it.
+        if (_be_exec_version >= 2) {
+            for (size_t i = 0; i < sz; i++) {
+                auto ele = argument_column->get_data_at(i);
```
Comment (HappenLee) [inline be/src/vec/exprs/vbloom_predicate.cpp:99]: Add a todo here to do remove virtual function call in `get_data_at` to improve performance

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run clickbench

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**